### ISSUE TITLE
Redundant MonadIO/Member (Lift IO) constraints required in some Carrier instances

### ISSUE THREAD

**Issue by robrix** [robrix/fused-effects#82]
If you write a `Carrier` instance which needs to lift `IO` actions into the action, you can use a `MonadIO` constraint on `m` and then use `liftIO` as normal. However, if you also have to run an effect handler inside the `Carrier` instance, on an action which requires `MonadIO`, then you have to add a superfluous `Member (Lift IO) sig` constraint, because the effect handler is specified in terms of `Eff`, and `Eff`’s `MonadIO` instance is contingent on a `Member (Lift IO) sig` constraint. So even tho you have a `MonadIO` constraint for `m`, it doesn’t automatically mean you have a `MonadIO` instance for `Eff (Blah m)`.

We could elide the `MonadIO` constraint if we didn’t have to run `liftIO` beside the effect handler (i.e. in `m`); but sometimes you do have to. We could also elide it if we changed the `Carrier` instance to operate on `Whatever (Eff m)` instead of `Whatever m`. But that’s unidiomatic.

So instead maybe we should offer a `sendIO` function in `Control.Effect.Lift` which does the appropriate lifting for us:

```haskell
sendIO :: (Carrier sig m, Member (Lift IO) sig) => IO a -> m a
sendIO = send . Lift . fmap pure
```

(This is the same definition as `liftIO` for `Eff`.)

We could also generalize it to `sendM`:

```haskell
sendM :: (Carrier sig m, Member (Lift n) sig, Monad n) => n a -> m a
sendM = send . Lift . fmap pure
```

(Technically we could loosen the constraint on `n` to `Functor`, but you wouldn’t be able to run the carrier without a `Monad` instance so there’s not much point in loosening there, AFAICS.)

Note however that similar problems (and potentially similar solutions) exist for the other effects whose only smart constructors are defined using instances on `Eff`, e.g. `Fail`/`MonadFail`, `NonDet`/`Alternative`, &c.

**Comment by patrickt** [comment]
I think adding `sendIO` and  as “equal to `liftIO`, but with better type inference in certain handlers” would be a fine addition.

**Comment by robrix** [comment]
Per discussion in #98 & #99, a general `sendM` seems like a good idea too.

**Comment by robrix** [comment]
This was fixed in #111: `MonadIO` is now the authoritative interface, so no `Member (Lift IO) sig` constraint is necessary.
